### ISSUE TITLE
_get_workgroup_config function does not acquire s3_output of primary group when workgroup is not given

### ISSUE THREAD

**Issue by Kit-Ok** [aws/aws-sdk-pandas#2518]
### Describe the bug
Setting AWS Athena 'primary' work group to force the S3 path to output, and execute
```python
wr.athena.read_sql_table(table_name, database, boto3_session=session)
```
and the error raised
```
botocore.errorfactory.InvalidRequestException: An error occurred (InvalidRequestException) when calling the StartQueryExecution operation: The Create Table As Select query failed because it was submitted with an 'external_location' property to an Athena Workgroup that enforces a centralized output location for all queries. Please remove the 'external_location' property and resubmit the query.
```

# reason
The `primary` workgroup is implicitly used when `workgroup` is not given, but in the [wr.athena._utils._get_workgroup_config](https://github.com/aws/aws-sdk-pandas/blob/24e6e81270f426072e54ab5336d5a022c8537a19/awswrangler/athena/_utils.py#L146C5-L146C30) function, when workgroup is not given, `_WorkGroupConfig` will be `_WorkGroupConfig(enforced=False, s3_output=None, encryption=None, kms_key=None)`.

### How to Reproduce

Setting AWS Athena 'primary' work group to force the S3 path to output, and execute
```python
import boto3
import awswrangler as wr

session = boto3.Session(profile_name='your_aws_profile')

table_name = "your_table_name"
database = "your_database"

wr.athena.read_sql_table(table_name, database, boto3_session=session)
```
and the error raised
```
botocore.errorfactory.InvalidRequestException: An error occurred (InvalidRequestException) when calling the StartQueryExecution operation: The Create Table As Select query failed because it was submitted with an 'external_location' property to an Athena Workgroup that enforces a centralized output location for all queries. Please remove the 'external_location' property and resubmit the query.
```

Just to check the behavior of `wr.athena._utils._get_workgroup_config` function,
```python
import boto3
import awswrangler as wr

session = boto3.Session(profile_name='your_aws_profile')

print(wr.athena._utils._get_workgroup_config(session=session))
print(wr.athena._utils._get_workgroup_config(session=session, workgroup='primary'))
```

### Expected behavior

get primary workgroup config when `workgroup` variable is not set, or force to set `workgroup` explicitly.

### OS

macOS Sonoma

### Python version

3.10.13

### AWS SDK for pandas version

3.4.1

**Comment by kukushking** [comment]
HI @Kit-Ok thanks for opening this. Yes we should fetch the primary workgroup details if workgroup is not given. Will open a PR for this.

**Comment by kukushking** [comment]
Fixed in #2526.
